internal/fsutil/create.go

```diff
@@ -121,6 +128,28 @@ func createSymlink(o *CreateOptions) error {
 	return os.Symlink(o.Link, o.Path)
 }
 
+func createHardLink(o *CreateOptions) error {
+	debugf("Creating hard link: %s => %s", o.Path, o.Link)
+	targetInfo, err := os.Lstat(o.Link)
+	if err != nil && os.IsNotExist(err) {
+		return fmt.Errorf("the target file does not exist: %s", o.Link)
+	} else if err != nil {
+		return err
+	}
+
+	linkInfo, err := os.Lstat(o.Path)
+	if err == nil || os.IsExist(err) {
+		if os.SameFile(targetInfo, linkInfo) {
+			return nil
+		}
+		return fmt.Errorf("the link already exists: %s", o.Path)
```

Comment:
What about `path %s already exists` (check the rest of the code to see if we use %s or %q for paths, I don't remember)